### DISCUSSION TITLE
Notifications received while the Android device is turned off are received when the device is turned on.

### DISCUSSION THREAD

**Discussion by matsuda-argano** [invertase/react-native-firebase#6773] [Q&A] [answered]
First of all thank you for the great library. I appreciate it.

Please note that I am using a translation software and the English may not be correct.

I am creating a VOIP app using "@react-native-firebase/messaging" for incoming call notifications. I am displaying a notification when there is an incoming call.

The problem occurs when an incoming call is received while the Android device is turned off.
When the Android device is turned on, the onReceive method of the ReactNativeFirebaseMessagingReceiver class is called.
This causes the incoming call notification to be displayed.

My question is: Is it a specification that the onReceive method is called when the Android device is turned on if the notification is received while the Android device is turned off?

**Comment by mikehardy** [comment, marked as answer]
Firebase Cloud Messages are queued for delivery based on the TTL of the message.

So if the TTL is high and the device is off, then yes when the device is on again the FCM will de-queue and spool out to the device, triggering handlers

https://firebase.google.com/docs/cloud-messaging/concept-options#lifetime

> When an app server posts a message to FCM and receives a message ID back, it does not mean that the message was already delivered to the device. Rather, it means that it was accepted for delivery. What happens to the message after it is accepted depends on many factors.

and later 

> If the device is not connected to FCM, the message is stored until a connection is established (again respecting the collapse key rules). When a connection is established, FCM delivers all pending messages to the device. If the device never gets connected again (for instance, if it was factory reset), the message eventually times out and is discarded from FCM storage. The default timeout is four weeks, unless the time_to_live flag is set.

Here is specific discussion on how to limit the lifespan of a message, referencing the case of an incoming call even:

https://firebase.google.com/docs/cloud-messaging/concept-options#ttl

**Reply by matsuda-argano** [reply]
Thank you for your very thorough and clear explanation.
